### PR TITLE
Encoding error appears when iterable is passed in query args list.

### PR DESCRIPTION
**Pull request by zhiltsov-sigma**
To reproduce the issue put the list which contains two types of items: iterable containing unicode string and unicode string with non-ascii symbol

### CODE REVIEW

**Comment by szsigmaukraine** [comment]
Hi, @methane.
What about this PullRequest?
Is it OK?

**Comment by methane** [comment]
I'll check it carefully when I have time, since there is risk of compatibility.
FYI: https://github.com/farcepest/MySQLdb1/issues/45

**Comment by methane** [comment]
and #200

**Comment by szsigmaukraine** [comment]
I modified a little bit test so now in tests following edge cases:
1. unicode arguments in the execute parameters (self-detected issue)
2. backward compatibility issue (farcepest/MySQLdb1#45)
3. single-element iterables in args also in dict-args ([farcepest/MySQLdb1/pull/36](https://github.com/farcepest/MySQLdb1/pull/36))

File: pymysql/tests/test_issues.py [hunk 1]
```diff
@@ -381,3 +383,30 @@ def test_issue_175(self):
...
+            self.assertEqual(cursor.execute(sql_insert_template, data[0]), 1)
+            self.assertEqual(cursor.execute(sql_insert_template, data[1]), 1)
+            self.assertEqual(cursor.execute(sql_dictionary_insert_template, data[2]), 1)
+            self.assertEqual(cursor.execute(sql_select_template, [(u'a', u'b', u'c'), u'\u0430']), 3)
+            self.assertEqual(cursor.fetchone(), (u'a', u'\u0430'))
+            self.assertEqual(cursor.fetchone(), (u'b', u'\u0430'))
+            self.assertEqual(cursor.fetchone(), (u'c', u'\u0430'))
+        finally:
+            cursor.execute('drop table if exists %s' % table_name)
+            connection.close()
```
Comment (methane) [inline pymysql/tests/test_issues.py]: Text files must end with LF.

File: pymysql/tests/test_issues.py [hunk 2]
```diff
@@ -381,3 +383,30 @@ def test_issue_175(self):
                     warnings.filterwarnings("ignore")
                     cur.execute('drop table if exists test_field_count')
 
+    def test_pull_request_for_fix_tuple_and_unicode_encoding_issue(self):
+        """ Encoding error appears when iterable is passed in query args list.
+        List should contain two types of items:
+        iterable containing unicode string and unicode string with non-ascii symbol"""
+        table_name = 'tuple_and_unicode'
```
Comment (methane) [inline pymysql/tests/test_issues.py]: Don't do this and write table name in SQL directly.

File: pymysql/tests/test_issues.py [hunk 3]
```diff
@@ -381,3 +383,30 @@ def test_issue_175(self):
...
+        connection = pymysql.connect(charset="utf8", **self.databases[0])
```
Comment (methane) [inline pymysql/tests/test_issues.py]: use `conn = self.connections[0]`.

File: pymysql/tests/test_issues.py [hunk 4]
```diff
@@ -381,3 +383,30 @@ def test_issue_175(self):
...
+        cursor = connection.cursor()
+        try:
+            cursor.execute('create table %s (value_1 varchar(1), value_2 varchar(1))' % table_name)
```
Comment (methane) [inline pymysql/tests/test_issues.py]: Use `self.create_table()`. See test_issue_33.

File: pymysql/tests/test_issues.py [hunk 5]
```diff
@@ -381,3 +383,30 @@ def test_issue_175(self):
                     warnings.filterwarnings("ignore")
                     cur.execute('drop table if exists test_field_count')
 
+    def test_pull_request_for_fix_tuple_and_unicode_encoding_issue(self):
+        """ Encoding error appears when iterable is passed in query args list.
```
Comment (methane) [inline pymysql/tests/test_issues.py]: Too long name. "pull_request_for_fix" is too verbose.
Maybe, "test_encoding_tuple_of_unicode_and_bytes"?

Comment (methane) [inline pymysql/tests/test_issues.py]: And could you add link to this issue in docstring?

Comment (szsigmaukraine) [inline pymysql/tests/test_issues.py]: Thnx for feedback.
I'll apply your comments on next week.

Sent from my HTC

----- Reply message -----

File: pymysql/tests/test_issues.py (hunk 4 above)
Comment (szsigmaukraine) [inline pymysql/tests/test_issues.py]: I did not found `self.create_table()` in the test_issue_33
Did you mean DatabaseTest.create_table(). So could I inherit TestGitHubIssues from this class or move my test into it?

**Comment by hannosch** [comment]
The changes in this PR have been committed as part of b46f59b9339bee0562d22ca0631bf592b1da6b67. The actual code change is in https://github.com/PyMySQL/PyMySQL/commit/b46f59b9339bee0562d22ca0631bf592b1da6b67#diff-5babc7bd89db074d50c856d6cd4965f2R102 and the new test (with PR feedback addressed) in https://github.com/PyMySQL/PyMySQL/commit/b46f59b9339bee0562d22ca0631bf592b1da6b67#diff-ad807e5133c4af924c8c19a42f1ad31aR380.

I think the PR can now be closed.

**Comment by methane** [comment]
Thanks.
